cpp/daal/src/algorithms/svm/svm_train_thunder_cache.h

```diff
@@ -105,107 +104,115 @@ class SVMCache<thunder, noCache, algorithmFPType, cpu> : public SVMCacheIface<th
         return SVMCachePtr<thunder, algorithmFPType, cpu>(res);
     }
 
-    services::Status getRowsBlock(const uint32_t * indices, algorithmFPType *& block) override
+    services::Status getRowsBlock(const uint32_t * const indices, const size_t n, NumericTablePtr & block) override
     {
         services::Status status;
-
-        uint32_t * indicesNew = const_cast<uint32_t *>(indices);
-        if (_isComputeSubKernel)
+        auto kernelResultTable   = SOANumericTableCPU<cpu>::create(n, _lineSize, DictionaryIface::FeaturesEqual::equal, &status);
+        size_t nIndicesForKernel = 0;
         {
-            indicesNew = indicesNew + _nSelectRows;
-        }
-
-        DAAL_CHECK_STATUS(status, _blockTask->copyDataByIndices(indicesNew, _xTable));
-        DAAL_ITTNOTIFY_SCOPED_TASK(cacheCompute);
-        DAAL_CHECK_STATUS(status, _kernel->computeNoThrow());
-        block = _cache.get();
-        return status;
-    }
+            DAAL_ITTNOTIFY_SCOPED_TASK(cache.getRowsBlock);
 
-    services::Status copyLastToFirst() override
-    {
-        DAAL_ITTNOTIFY_SCOPED_TASK(cache.copyLastToFirst);
-
-        _nSelectRows = _cacheSize / 2;
-        if (!_isComputeSubKernel)
+            for (int i = 0; i < n; ++i)
```

Comment:
seems extra space at the beginning of the line, please check it (apply alignment for all lines bellow)

Reply:
No. Git displays crookedly. the space is correct. See the full file.